### PR TITLE
Fix and optimize the test framework + fix subtle bug in Parsec

### PR DESCRIPTION
**Pull request by madadam**
1. Fix the `Parsec::previous_interesting_content` function to return `None` in case the previous interesting content is empty. In that case we must perform the full computation, because there can be new interesting content that wasn't present in the previous meta-election.
2. Make `PeerRemovalGuard` more robust to handle edge cases.
3. Lower the gossip probability to speed the tests up, but increase the `adjustment_coeff` correspondingly, to make sure they still pass. Overall significant test performance increase.
4. Remove `prob_local_step` as it wasn't really used anymore (local step now happens on every step).
5. Some minor refactoring of the test framework

### CODE REVIEW

File: src/dev_utils/network.rs [hunk 1]
```diff
@@ -544,34 +534,115 @@ impl Network {
...
+// Helper struct that protects the test network from losing too many nodes.
 #[derive(Default)]
 struct PeerRemovalGuard {
-    voters: BTreeSet<PeerId>,
-    nodes_to_be_removed: BTreeSet<PeerId>,
+    peers: BTreeMap<PeerId, PeerRemovalState>,
+}
+
+enum PeerRemovalState {
+    Adding(usize),
```
Comment (maqi) [inline src/dev_utils/network.rs]: better have comment here to explain the `usize` is for the count of voters
